kernel/mutex.c

```diff
@@ -66,10 +67,61 @@ int __myst_mutex_trylock(myst_mutex_t* m, myst_thread_t* self)
     return -EBUSY;
 }
 
+typedef struct myst_mutex_thread_sig_handler
+{
+    // Used by thread signal handler infrastructure
+    myst_thread_sig_handler_t sig_handler;
+
+    // our mutex details to clean up
+    myst_mutex_t* mutex;
+
+} myst_mutex_thread_sig_handler_t;
+
+/* Our default handler just needs to remove itself from the queue and call the
+ * next in line */
+static void myst_mutex_sig_handler(
+    MYST_UNUSED unsigned signum,
+    void* _sig_handler)
+{
+    myst_mutex_thread_sig_handler_t* sig_handler =
+        (myst_mutex_thread_sig_handler_t*)_sig_handler;
+    myst_thread_t* thread = myst_thread_self();
+
+    myst_spin_lock(&sig_handler->mutex->lock);
+
+    myst_thread_queue_remove_thread(&sig_handler->mutex->queue, thread);
+
+    thread->signal.waiting_on_event = false;
+
+    if (sig_handler->mutex->queue.front != NULL)
```

Comment:
Should we do the wake only if there is no `owner` for the mutex? The owner should be responsible for waking the next waiter.